fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -3854,7 +3855,17 @@ public ParseNode visitDropFrontendClause(StarRocksParser.DropFrontendClauseConte
     public ParseNode visitModifyFrontendHostClause(StarRocksParser.ModifyFrontendHostClauseContext context) {
```

Comment:
More a question to better my understanding- Where does the call for this visitor method originate?

Reply:
`visitModifyFrontendHostClause` is called by `ModifyFrontendHostClauseContext.accept`.  `ModifyFrontendHostClauseContext` is an automatically generated subclass of `org.antlr.v4.runtime.ParseRuleContext`, and therefore the interface between the antlr4 parser and the user-written StarRocks FE code. My understanding is that antlr4 is invoked to parse the input SQL statement into a syntax tree, and the `<ParseTreeVisitor>.accept` (in this case, `AstBuilder`) is where the user-written starrocks code begins.

Reply:
Thanks for the context. I think my confusion was from the transition from visitModifyFrontendHostClause --> visitModifyFrontendClause.